parser/parser.go

```diff
@@ -17,35 +17,127 @@
 
 package parser
 
-import "errors"
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/datastax/go-cassandra-native-protocol/datatype"
+	"github.com/datastax/go-cassandra-native-protocol/message"
+	"github.com/google/uuid"
+)
 
 type Selector interface {
-	isSelector()
+	Values(columns []*message.ColumnMetadata, valueFunc ValueLookupFunc) (filtered []message.Column, err error)
+	Columns(columns []*message.ColumnMetadata, stmt *SelectStatement) (filtered []*message.ColumnMetadata, err error)
```

Comment:
Even if this weren't fixing the underlying problem the conversion to managing values and columns within selectors rather than as external util functions is worth the cost of admission by itself.